src/reservoir/utils.ts

```diff
@@ -0,0 +1,348 @@
+  startTime: bigint;
+  endTime: bigint;
+  zoneHash: Hash;
+  salt: bigint;
+  conduitKey: Hash;
+  counter: bigint;
+}
+
+export interface SeaportOrderParameter extends SeaportOrder {
+  totalOriginalConsiderationItems: bigint;
+}
+
+export interface FulfillmentComponent {
+  orderIndex: bigint;
+  itemIndex: bigint;
+}
+
+export interface Fulfillment {
+  offerComponents: FulfillmentComponent[];
+  considerationComponents: FulfillmentComponent[];
+}
+
+export const ETH_CONTRACT_ADDRESS = zeroAddress;
+export const WETH_CONTRACT_ADDRESS =
+  "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2";
+const SEAPORT_CONTRACT_NAME = "Seaport";
+const SEAPORT_VERSION = "1.5";
+const SEAPORT_CONTRACT_ADDRESS =
+  "0x00000000000000ADc04C56Bf30aC9d3c0aAF14dC" as const;
+
+// We don't have this abi in the generated files
+export const EXECUTION_INFO_ABI = [
+  {
+    name: "ExecutionInfo",
+    type: "tuple",
+    components: [
+      { name: "module", type: "address" },
+      { name: "data", type: "bytes" },
+      { name: "value", type: "uint256" },
+    ],
+  },
+] as const;
+
+export const getExecutionData = ({
+  module,
+  data,
+  value,
+}: {
+  module: Address;
+  data: Hash;
+  value: bigint;
+}) => encodeAbiParameters(EXECUTION_INFO_ABI, [{ module, data, value }]);
+
+export const adaptWalletToCaptureTxData = (wallet: Wallet) => {
+  // Here we will store the tx data we want to capture
+  const txData = {
+    orderId: zeroHash,
+    callbackData: zeroHash,
+    to: zeroAddress as Address,
+    value: 0n,
+    signature: zeroHash,
+  };
+
+  const viemWallet = adaptViemWallet(wallet);
+
+  const adaptedWallet = {
+    ...viemWallet,
+    transport: undefined,
+    handleSendTransactionStep: async (
+      _chainId: number,
+      stepItem: {
+        data: { data: Hash; to: Address; value: bigint };
+        orderIds: Hash[];
+      },
+    ) => {
+      txData.orderId = stepItem.orderIds[0];
+      txData.to = stepItem.data.to;
+      txData.callbackData = stepItem.data.data;
+      txData.value = stepItem.data.value;
+      try {
+        // We try to decode the function data to know if it's a seaport contract call
+        // If it is, we can save a tx by using matchOrders method from the seaport contract
+        // If it's not, we use the same tx data
+        const functionData = decodeFunctionData({
+          abi: seaportABI,
+          data: stepItem.data.data,
+        });
+
+        // seaport contract calls have a signature in the argument
+        if (
+          functionData.args &&
+          functionData.args[0] &&
+          typeof functionData.args[0] === "object" &&
+          "signature" in functionData.args[0]
+        ) {
+          txData.signature = functionData.args[0].signature;
+        }
+      } catch {
+        // We ignore the error if the tx is not a seaport tx
+      }
+
+      // We throw the error to stop the tx, we will do it inside the Leverage contract
+      throw new Error("We don't want the tx to continue");
```

Comment:
I'd throw custom error here. Something like [this](https://gomycode.com/eg/error-handling-in-javascript-custom-errors/)